data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt

```diff
@@ -1,28 +1,43 @@
 package io.gnosis.data.repositories
 
+import io.gnosis.data.backend.GatewayApi
 import io.gnosis.data.backend.TransactionServiceApi
-import io.gnosis.data.backend.dto.ServiceTokenInfo
-import io.gnosis.data.backend.dto.tokenAsErc20Token
-import io.gnosis.data.db.daos.Erc20TokenDao
-import io.gnosis.data.models.Balance
-import io.gnosis.data.models.Collectible
-import io.gnosis.data.models.Erc20Token
+import io.gnosis.data.models.*
 import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
 import pm.gnosis.model.Solidity
 import java.math.BigInteger
 
 class TokenRepository(
-    private val erc20TokenDao: Erc20TokenDao,
-    private val transactionServiceApi: TransactionServiceApi
+    private val transactionServiceApi: TransactionServiceApi,
+    private val gatewayApi: GatewayApi
 ) {
 
-    suspend fun loadBalanceOf(safe: Solidity.Address): List<Balance> =
-        transactionServiceApi.loadBalances(safe.asEthereumAddressChecksumString())
-            .map {
-                val token = it.tokenAsErc20Token()
-                erc20TokenDao.insertToken(token)
-                Balance(token, it.balance, it.balanceUsd)
-            }
+    suspend fun loadBalanceOf(safe: Solidity.Address): CoinBalances {
+        val response = gatewayApi.loadBalances(safe.asEthereumAddressChecksumString())
+        return CoinBalances(response.fiatTotal, response.items.map {
+
+            if (it.tokenInfo.address == null)
```

Comment:
It's good that you don't rely on the `TokenInfo::type` here, there is been some discussion about that and there might be changes in the API for the type (because of xDai) this will definitely make things easier. I hope iOS also doesn't use `TokenType::Ether`